src/main/java/com/teragrep/pth_06/planner/walker/conditions/ElementCondition.java

```diff
@@ -133,16 +135,37 @@ public Condition condition() {
 
     @Override
     public boolean equals(final Object object) {
-        if (this == object)
+        if (this == object) {
             return true;
-        if (object == null)
+        }
+        if (object == null) {
             return false;
-        if (object.getClass() != this.getClass())
+        }
+        if (object.getClass() != this.getClass()) {
             return false;
+        }
         final ElementCondition cast = (ElementCondition) object;
         boolean equalName = this.element.getTagName().equals(cast.element.getTagName());
         boolean equalOperation = this.element.getAttribute("operation").equals(cast.element.getAttribute("operation"));
         boolean equalValue = this.element.getAttribute("value").equals(cast.element.getAttribute("value"));
         return equalName && equalOperation && equalValue && this.config.equals(cast.config);
     }
+
+    @Override
+    public int hashCode() {
```

Comment:
hashcode

Reply:
Solved in commit 4d4e0023426356b05c592147f65933f69ba5dfa3